src/message_dao.py

```diff
@@ -109,6 +109,15 @@ def get(self, message_id: str) -> Optional[Message]:
         msg.children = self._get_children(message_id)
         return msg
 
+    @lru_cache()
+    def previous_version_candidates(self, message: Message) -> List[Message]:
+        reduced_subject = message.subject.partition("] ")[2].lower()
+        sql = "SELECT message_id FROM Messages WHERE from_=%s AND reduced_subject=%s"
```

Comment:
note: should we require a non-empty change id?
Since we only really care about the previous versions for the change id anyway.

And it would guard against some edge cases:
What happens if someone replies to a patch but they don't change the subject to start with "Re: "?
Or what happens if we just fail to apply the parent patch (since it doesn't apply to torvalds/master), but the child does? (Answer, we could publish v2 as a patchset 1, which could be confusing...)
Or what happens if they send a malformed buggy version of a patch and then resend again with the same subject and version #?

Reply:
Query now excludes messages where the change id is empty. Alternate solution to consider might be excluding the existence of null change ids within the database; that is, not allowing the `change_id` column to be a null value.